Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -55,14 +55,19 @@ bool VirtualFileSystem::mount_point_exists_at_inode(InodeIdentifier inode_identi
 {
     return m_mounts.with([&](auto& mounts) -> bool {
         return any_of(mounts, [&inode_identifier](auto const& existing_mount) {
-            return existing_mount->host() && existing_mount->host()->identifier() == inode_identifier;
+            return existing_mount.host() && existing_mount.host()->identifier() == inode_identifier;
         });
     });
 }
 
 ErrorOr<void> VirtualFileSystem::mount(FileSystem& fs, Custody& mount_point, int flags)
 {
-    auto new_mount = TRY(adopt_nonnull_own_or_enomem(new (nothrow) Mount(fs, &mount_point, flags)));
+    auto new_mount = new (nothrow) Mount(fs, &mount_point, flags);
+    if (!new_mount)
+        return Error::from_errno(ENOMEM);
+    ArmedScopeGuard delete_mount_on_failure { [&] {
+        delete new_mount;
+    } };
```

Comment:
This looks much more complicated than it needs to be. Can we use `OwnPtr` instead and let the `IntrusiveList` own the `Mount` instances?

Reply:
Look at the TmpFS code, it does the same for inode children, because trying to force to use `OwnPtr` results in the error:
```
AK/IntrusiveList.h:229:32: error: use of deleted function 'AK::OwnPtr<T>& AK::OwnPtr<T>::operator=(T*) [with T = Kernel::Mount]'
  229 |         nnode.m_self.reference = &n; // Note: Self-reference ensures that the object will keep a ref to itself when the Container is a smart pointer.
```
I try to see if there's a clever trick to do here so IntrusiveLists could be used with OwnPtrs, not sure there's.

Reply:
So instead of trying to use `OwnPtr`s, and the fact that we still need to ensure that in case of an error that we delete the new mount - I added a small call to `leak_ptr` of the `NonnullOwnPtr` of the mount, so it basically does the same thing more elegantly :)